pkgs/development/interpreters/elixir/generic-builder.nix

```diff
@@ -22,10 +22,8 @@ in
 
     buildInputs = [ erlang rebar makeWrapper ];
 
-    LOCALE_ARCHIVE = stdenv.lib.optionalString stdenv.isLinux
```

Comment:
Are you sure this isn't necessary anymore? It as added to fix #30047.

/cc @gleber

Reply:
I have tried reproducing the problem in #30047 and the problem is gone. This change LGTM

Reply:
I'm sure also, even if it's not jet tested on my side.
This isn't necessary anymore since https://github.com/NixOS/nixpkgs/pull/54485 is merged.

With Elixir running within a `nix-shell` it is even enought to just set `LANG` - without `LC_TYPE`.
But as I once have read Elixir also looks on `LC_TYPE` I would leave it here for now.
We could test this and even remove `LANG` when https://github.com/NixOS/nixpkgs/pull/61202 is merged.